src/services/interestsService.js

```diff
@@ -0,0 +1,19 @@
+export default function InterestsService() {
+  return {
+    getCommonInterests: (interests1, interests2) => {
```

Comment:
Algo [así](https://stackoverflow.com/questions/34392741/best-way-to-get-intersection-of-keys-of-two-objects/36960575#36960575) funciona también:
```js
return Object.keys(interests1).filter(interest => interest in interests2)
```

Que te devuelve un array con los `id` que coinciden, asique en los tests sólo podrías comparar por `id` y no por `name`

Reply:
pasa que así es cuartico y como yo lo hice es lineal

Reply:
Por qué cuártico?

Reply:
por que para cada interes de usuario 1, mira todos los del usuario 2

Reply:
Mmm lo decís por el `in`? No sé qué complejidad tiene la verdad, pero si es `O(n)` sería cuadrático en todo caso. En caso de que haga algo parecido a lo que hacés vos, sería lo mismo, decía porque era más prolijo jajaj pero 👍

Reply:
Siga siga